BondageClub/Scripts/ImageCache.js

```diff
@@ -0,0 +1,217 @@
+"use strict";
+
+/**
+ * What state is the image in.
+ * @enum string
+ */
+const CachedImageState = {
+	UNCACHED: 'uncached',
+	LOADING: 'loading',
+	LOADED: 'loaded',
+	ERRORED: 'errored',
+	FAILED: 'failed',
+};
+
+/**
+ * Cached image data
+ */
+class CachedImage {
+	/**
+	 * Create a new cached image.
+	 *
+	 * @param {ImageCache} cache - The cache holding the image
```

Comment:
I think this type should be `ImageCacheClass`
```suggestion
	 * @param {ImageCacheClass} cache - The cache holding the image
```